test/unit/loan/SimplePSM.t.sol

```diff
@@ -0,0 +1,181 @@
+        vm.label(address(token), "token");
+        vm.label(address(credit), "credit");
+        vm.label(address(guild), "guild");
+        vm.label(address(rlcm), "rlcm");
+        vm.label(address(psm), "psm");
+        vm.label(address(this), "test");
+    }
+
+    // constructor params & public state getters
+    function testInitialState() public {
+        assertEq(address(psm.core()), address(core));
+        assertEq(psm.profitManager(), address(profitManager));
+        assertEq(psm.rlcm(), address(rlcm));
+        assertEq(psm.credit(), address(credit));
+        assertEq(psm.token(), address(token));
+        assertEq(psm.decimalCorrection(), 1e12);
+    }
+
+    // enforce that:
+    // - decimal normalization occur properly betwheen input & output
+    // - creditMultiplier is taken into account
+    function testGetMintAmountOut() public {
+        assertEq(psm.getMintAmountOut(0), 0);
+        assertEq(psm.getMintAmountOut(123e6), 123e18);
+
+        // update creditMultiplier
+        credit.mint(address(this), 100e18);
+        assertEq(profitManager.creditMultiplier(), 1e18);
+        profitManager.notifyPnL(address(this), -50e18);
+        assertEq(profitManager.creditMultiplier(), 0.5e18);
+
+        assertEq(psm.getMintAmountOut(0), 0);
+        assertEq(psm.getMintAmountOut(100_000_000), 2 * 100_000_000 * 1e12);
+        assertEq(psm.getMintAmountOut(111), 222e12);
+    }
+
+    // enforce that:
+    // - decimal normalization occur properly betwheen input & output
+    // - creditMultiplier is taken into account
+    // - rounding errors due o creditMultiplier are as expected
+    function testGetRedeemAmountOut() public {
+        assertEq(psm.getRedeemAmountOut(0), 0);
+        assertEq(psm.getRedeemAmountOut(123e18), 123e6);
+
+        // update creditMultiplier
+        credit.mint(address(this), 100e18);
+        assertEq(profitManager.creditMultiplier(), 1e18);
+        profitManager.notifyPnL(address(this), -50e18);
+        assertEq(profitManager.creditMultiplier(), 0.5e18);
+
+        assertEq(psm.getRedeemAmountOut(0), 0);
+        assertEq(psm.getRedeemAmountOut(2 * 100_000_000 * 1e12), 100_000_000);
+        assertEq(psm.getRedeemAmountOut(111_111_111_111_111_111_111), 55_555_555);
+        assertEq(psm.getRedeemAmountOut(12345), 0);
+    }
+
+    // enforce that:
+    // - mint moves a number of tokens equal to getMintAmountOut() i/o
+    // - redeem moves a number of token equal to getRedeemAmountOut() i/o
+    // - rlcm buffer is depleted & replenished
+    // - rounding errors are within 1 wei for a mint/redeem round-trip
+    function testMintRedeem(uint256 input) public {
+        uint256 mintIn = input % 1e15 + 1; // [1, 1_000_000_000e6]
+
+        // update creditMultiplier
+        // this ensures that some fuzz entries will result in uneven divisions
+        // and will create a min/redeem round-trip error of 1 wei.
+        credit.mint(address(this), 100e18);
```

Comment:
why mint if the next step is just going to burn on 147?

Reply:
to update the `creditMultiplier` the totalSupply needs to be != 0, so I'm minting a bunch of tokens, updating the `creditMultiplier`, and then burning them